### PR TITLE
Improve the Visual Studio dev experience

### PR DESCRIPTION
**Pull request by Porges**
This allows you to run the tests in Visual Studio using the built-in Test Explorer window.

image

Doing this required taking the explicit dependency on FSharp.Core 4.0.0.1, since we have to have a version in the output directory of the test project for them to be able to run. I've made the dependency have a lower bound but no upper, so it won't force any particular version onto the consumers of the nuget package. (I was going to do this anyway for the C# bits.)

I've also included the various paket files into the solution to make it easier to view them.

### CODE REVIEW

**Comment by Porges** [comment]
Wow, that image came out bigger than expected 😀

File: tests/Hedgehog.Tests/Hedgehog.Tests.fsproj [hunk 1]
```diff
@@ -33,29 +33,6 @@
...
-    <Content Include="App.config" />
-  </ItemGroup>
-  <ItemGroup>
-    <ProjectReference Include="..\..\src\Hedgehog\Hedgehog.fsproj">
-      <Name>Hedgehog</Name>
-      <Project>{b338b8c4-1c98-4f42-9c45-56bd7ef2ad0a}</Project>
-      <Private>True</Private>
-    </ProjectReference>
-  </ItemGroup>
   <PropertyGroup>
```
Comment (moodmosaic) [inline tests/Hedgehog.Tests/Hedgehog.Tests.fsproj]: I guess these items where moved automatically? It's hard to spot if there are differences here on not 😕

File: src/Hedgehog/Hedgehog.fsproj [hunk 2]
```diff
@@ -18,32 +18,27 @@
...
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.0\Framework\v4.0\Microsoft.FSharp.Targets</FSharpTargetsPath>
   </PropertyGroup>
-  <ItemGroup>
-    <Reference Include="mscorlib" />
-    <Reference Include="FSharp.Core, Version=$(TargetFSharpCoreVersion), Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="System" />
-    <Reference Include="System.Core" />
-    <Reference Include="System.Numerics" />
```
Comment (moodmosaic) [inline src/Hedgehog/Hedgehog.fsproj]: I guess these items where moved automatically? It's hard to spot if there are differences here on not 😕

File: paket.dependencies [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
+// We want to depend on FSharp.Core 4.0.0.1, but not require
+// that version in the generated nuget package (see paket.template).
+//
+// If we just specify "FSharp.Core 4.0.0.1" then the nuget package
+// has an dependency on that *exact* version.
+//
+// So, we specify 4.0.0.1 as a minimum bound, and set lowest_matching
+// so it won't be updated automatically to a newer one by 'paket update'.
+nuget FSharp.Core >= 4.0.0.1 lowest_matching: true
+
```
Comment (moodmosaic) [inline paket.dependencies]: 👍 That's pretty awesome.

File: Hedgehog.sln [hunk 4]
```diff
@@ -1,9 +1,17 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2012
-Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Hedgehog", "src\Hedgehog\Hedgehog.fsproj", "{B338B8C4-1C98-4F42-9C45-56BD7EF2AD0A}"
+# Visual Studio 15
```
Comment (moodmosaic) [inline Hedgehog.sln]: I think this change was made automatically by Visual Studio(?)

**Comment by moodmosaic** [comment]
I think this is helpful to those who run their tests using the Visual Studio built-in test runner. After all, these tests are written using xUnit.net, so I don't see why not taking up a dependency on its Visual Studio runner.

Would it be possible to move those lines in the *.fsproj back to their original position? (See [this](https://github.com/hedgehogqa/fsharp-hedgehog/pull/90#discussion_r111318725) and [this](https://github.com/hedgehogqa/fsharp-hedgehog/pull/90/files/5879a9e0029cee507d30daaa4dca5708ebe1ed0e#diff-726cfb13c07cca41c2b35d5eea95b8ed) comment.)

**Comment by Porges** [comment]
It was actually paket that moved all the lines around. It seems to want to put them in the same place every time, so I think if they're moved once they won't move again in future...

File: Hedgehog.sln (hunk 4 above)
Comment (Porges) [inline Hedgehog.sln]: Yes. It shouldn't prevent older VS's from opening the solution.

File: paket.dependencies (hunk 3 above)
Comment (moodmosaic) [inline paket.dependencies]: FWIW, I just run a build with the NuGet flag to create a package locally. Indeed, it's now requiring FSharp.Core >= 4.0.0.1

image

👍

**Comment by jystic** [comment]
👍 thanks for the contribution, have rebased and merged

**Pull request closed without merging**